akka-stream/src/main/scala/akka/stream/impl/Stages.scala

```diff
@@ -93,112 +96,129 @@ private[stream] object Stages {
 
   import DefaultAttributes._
 
+  // FIXME: To be deprecated as soon as stream-of-stream operations are stages
   sealed trait StageModule extends FlowModule[Any, Any, Any] {
-    def attributes: Attributes
     def withAttributes(attributes: Attributes): StageModule
     override def carbonCopy: Module = withAttributes(attributes)
   }
 
-  final case class StageFactory(mkStage: () ⇒ Stage[_, _], attributes: Attributes = stageFactory) extends StageModule {
-    override def withAttributes(attributes: Attributes) = copy(attributes = attributes)
+  /*
+   * Stage that is backed by a GraphStage but can be symbolically introspected
+   */
+  case class SymbolicGraphStage[-In, +Out, Ext](symbolicStage: SymbolicStage[In, Out])
+    extends PushPullGraphStage[In, Out, Ext](
+      symbolicStage.create,
+      symbolicStage.attributes) {
   }
 
-  final case class MaterializingStageFactory(
-    mkStageAndMaterialized: () ⇒ (Stage[_, _], Any),
-    attributes: Attributes = stageFactory) extends StageModule {
-    override def withAttributes(attributes: Attributes) = copy(attributes = attributes)
+  val identityGraph = SymbolicGraphStage[Any, Any, Any](Identity)
+
+  sealed trait SymbolicStage[-In, +Out] {
```

Comment:
it depends what do you call a Stage. Only subclasses of `Stage`, or every atomic module with `FlowShape`. I usually use the latter, but it can be confusing here.